src/transport/GrpcTransport.js

```diff
@@ -0,0 +1,113 @@
+const {
+  CorePromiseClient,
+  PlatformPromiseClient,
+  TransactionsFilterStreamPromiseClient,
+} = require('@dashevo/dapi-grpc');
+
+const grpcErrorCodes = require('../errors/grpcErrorCodes');
+
+class GrpcTransport {
+  /**
+   * @param {MNDiscovery} mnDiscovery
+   * @param {number} dapiPort
+   * @param {number} grpcNativePort
+   * @param {string} clientType
+   */
+  constructor(mnDiscovery, dapiPort, grpcNativePort, clientType) {
+    this.mnDiscovery = mnDiscovery;
+    this.dapiPort = dapiPort;
+    this.grpcNativePort = grpcNativePort;
+    this.clientType = clientType;
+  }
+
+  /**
+   * Make request to a random MN (with retries)
+   *
+   * @param {string} method
+   * @param {Object} request
+   * @param {Object} options
+   * @param {number} [options.retriesCount=3]
+   * @param {string[]} [options.excludedIps=[]]
+   * @param {Object} [options.client={}]
+   * @param {number} [options.client.timeout]
+   *
+   * @returns {Promise<*|undefined>}
+   */
+  async makeRequest(method, request, options = { retriesCount: 3, excludedIps: [] }) {
+    const retriesCount = options.retriesCount || 3;
+    const excludedIps = options.excludedIps || [];
+
+    let urlToConnect;
+    try {
+      urlToConnect = await this.getGrpcUrl(excludedIps);
+
+      let client;
+      switch (this.clientType) {
+        case GrpcTransport.TYPES.CORE: {
+          client = new CorePromiseClient(urlToConnect);
```

Comment:
Pass the client constructor function as a dependency to `GrpcTransport` and remove switch.